tests/nvmeof/workflows/ha.py

```diff
@@ -932,11 +961,20 @@ def run(self):
                     log_json_dump(fail_method)
 
                     fail_gws, operational_gws = self.catogorize(nodes)
-                    fail_gw_ana_ids = [gw.ana_group_id for gw in fail_gws]
-                    gateway = operational_gws[0]
+                    fail_gw_ana_ids = []
+                    namespaces = []
+                    for gw in fail_gws:
+                        fail_gw_ana_ids.append(gw.ana_group_id)
+                        ns = self.fetch_namespaces(gw, [gw.ana_group_id])
+                        namespaces.extend(ns)
+                        self.validate_initiator(gw, ns, gw.ana_group_id)
```

Comment:
In validate_initiator,

1. At Initial stage, pass active gateway node object(which has to be failed over next).
2. Fetch namesapces for this GW from gateway and client side.
3. Validate optimized path(i.e, basically Active gateway which is passed).
4. At Failover stage, collect the active gateway from failover methods.
5. Use this Active GW against namespaces found in Step 2 to validate Optimized path on client is set to new active GW.
6. At failback condition, collect the failed-over gateway.
7. Repeat step 2.

Reply:
This is exactly the validation I have already added in the validate_initiator and ha.run methods